src/cascadia/TerminalSettingsModel/ActionArgs.idl

```diff
@@ -122,10 +122,22 @@ namespace Microsoft.Terminal.Settings.Model
         All = 0xffffffff,
     };
 
-    [default_interface] runtimeclass NewTerminalArgs {
+    interface INewContentArgs {
+        String Type { get; };
+        Boolean Equals(INewContentArgs other);
+        UInt64 Hash();
+        INewContentArgs Copy();
+        String GenerateName();
+    };
+
+    [default_interface] runtimeclass BaseContentArgs : INewContentArgs{
+        BaseContentArgs();
+        BaseContentArgs(String type);
+    };
+
+    [default_interface] runtimeclass NewTerminalArgs : INewContentArgs{

```

Comment:
pls stop throwing `[default_interface]` onto everything unless it actually NEEDS to have an interface!